File: app/src/main/java/com/hfad/myferma/WriteOff/WriteOffChartFragment.java
package com.hfad.myferma.WriteOff;

import android.database.Cursor;
import android.graphics.Color;
import android.os.Bundle;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.AutoCompleteTextView;
import android.widget.RadioButton;
import android.widget.RadioGroup;

import androidx.fragment.app.Fragment;

import com.github.mikephil.charting.charts.BarChart;
import com.github.mikephil.charting.components.XAxis;
import com.github.mikephil.charting.data.BarData;
import com.github.mikephil.charting.data.BarDataSet;
import com.github.mikephil.charting.data.BarEntry;
import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.formatter.IndexAxisValueFormatter;
import com.github.mikephil.charting.utils.ColorTemplate;
import com.google.android.material.appbar.MaterialToolbar;
import com.google.android.material.floatingactionbutton.ExtendedFloatingActionButton;
import com.hfad.myferma.R;
import com.hfad.myferma.db.MyFermaDatabaseHelper;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;

public class WriteOffChartFragment extends Fragment {
    private MyFermaDatabaseHelper myDB;
    private AutoCompleteTextView animals_spiner, mount_spiner, year_spiner;
    private ArrayList<BarEntry> visitors;
    private ArrayAdapter<String> arrayAdapterAnimals;

    private RadioButton radioButton1, radioButton2;

    private RadioGroup radioGroup;
    private String[] labes = {"", "Январь", "Февраль", "Март", "Апрель", "Май", "Июнь", "Июль", "Август", "Сентябрь", "Октябрь", "Ноябрь", "Декабрь", ""};
    private String[] mountMass;
    private int mount = 0;

    private int status = R.drawable.baseline_cottage_24;

    private View layout;

    private String infoChart = "График списанной продукции на собсвенные нужды";
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        layout = inflater.inflate(R.layout.fragment_write_off_chart, container, false);

        // установка радио
        radioGroup = (RadioGroup) layout.findViewById(R.id.radioGroup);
        radioButton1 = (RadioButton) layout.findViewById(R.id.radio_button_1);
        radioButton2 = (RadioButton) layout.findViewById(R.id.radio_button_2);

        // установка спинеров
        animals_spiner = (AutoCompleteTextView) layout.findViewById(R.id.animals_spiner);
        mount_spiner = (AutoCompleteTextView) layout.findViewById(R.id.mount_spiner);
        year_spiner = (AutoCompleteTextView) layout.findViewById(R.id.year_spiner);

        //Подключение к базе данных
        myDB = new MyFermaDatabaseHelper(getActivity());

        //Создание списка с данными для графиков
        visitors = new ArrayList<>();

        // настройка спинеров
        animals_spiner.setText("Яйца", false);
        mount_spiner.setText("За весь год", false);
        year_spiner.setText("2023", false);

        //убириаем фаб кнопку
        ExtendedFloatingActionButton fab = (ExtendedFloatingActionButton) getActivity().findViewById(R.id.extended_fab);
        fab.setVisibility(View.GONE);

        //настройка верхнего меню фаб кнопку
        MaterialToolbar appBar = getActivity().findViewById(R.id.topAppBar);
        appBar.setTitle("Мои Списания - График");

        //Логика просчета
        storeDataInArrays();

        // настройка графиков
        bar(labes,  infoChart);

        radioGroup.setOnCheckedChangeListener((group, checkedId) -> {
            switch (checkedId) {
                case R.id.radio_button_1:
                    visitors.clear();
                    status = R.drawable.baseline_cottage_24;
                    storeDataInArrays();
                    infoChart = "График списанной продукции на собсвенные нужды";
                    if (mount != 13) {
                        bar(mountMass,  infoChart);
                    } else {
                        bar(labes,  infoChart);
                    }
                    break;

                case R.id.radio_button_2:
                    visitors.clear();
                    status = R.drawable.baseline_delete_24;
                    storeDataInArrays();
                    infoChart = "График списанной продукции на утилизацию";
                    if (mount != 13) {
                        bar(mountMass,  infoChart);
                    } else {
                        bar(labes,  infoChart);
                    }
                    break;
            }
        });
        animals_spiner.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                visitors.clear();
                storeDataInArrays();
                if (mount != 13) {
                    bar(mountMass,  infoChart);
                } else {
                    bar(labes,  infoChart);
                }
            }
        });
        mount_spiner.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                visitors.clear();
                storeDataInArrays();
                if (mount != 13) {
                    bar(mountMass,  infoChart);
                } else {
                    bar(labes,  infoChart);
                }
            }
        });
        year_spiner.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                visitors.clear();
                storeDataInArrays();
                if (mount != 13) {
                    bar(mountMass,  infoChart);
                } else {
                    bar(labes,  infoChart);
                }
            }
        });
        return layout;
    }

    public void bar(String[] xAsis, String info) {
        BarChart barChart = layout.findViewById(R.id.barChart);
        BarDataSet barDataSet = new BarDataSet(visitors, animals_spiner.getText().toString());
        barDataSet.setColors(ColorTemplate.MATERIAL_COLORS);
        barDataSet.setValueTextColor(Color.BLACK);
        barDataSet.setValueTextSize(16f);

        BarData barData = new BarData(barDataSet);
        barChart.invalidate();
        barChart.setFitBars(true);
        barChart.setData(barData);
        barChart.getDescription().setText(info);
        barChart.animateY(2000);

        XAxis xAxis = barChart.getXAxis();
        xAxis.setPosition(XAxis.XAxisPosition.BOTTOM);
        xAxis.setDrawGridLines(false);
        xAxis.setGranularity(1f); // only intervals of 1 day
        xAxis.setLabelCount(6); //сколько отображается
        xAxis.setValueFormatter(new IndexAxisValueFormatter(xAsis));
    }
    @Override
    public void onStart() {
        super.onStart();
        View view = getView();
        if (view != null) {
            // настройка спинера с годами (выглядил как обычный, и год запоминал)
            arrayAdapterAnimals = new ArrayAdapter<>(getActivity().getApplicationContext(), android.R.layout.simple_spinner_dropdown_item, add());
            year_spiner.setAdapter(arrayAdapterAnimals);
        }
    }

    public ArrayList<String> add() {
        Set<String> tempList = new HashSet<>();
        Cursor cursor = myDB.readAllDataSale();

        while (cursor.moveToNext()) {
            String string1 = cursor.getString(5);
            tempList.add(string1);
        }
        cursor.close();

        ArrayList<String> tempList1 = new ArrayList<>();
        for (String nameExpenses : tempList) {
            tempList1.add(nameExpenses);
        }

        return tempList1;
    }

    void storeDataInArrays() {
        Cursor cursor = myDB.readAllDataWriteOff();

        Map<Float, Float> sumCategory = new HashMap<>();

        String animalsType = animals_spiner.getText().toString();
        String mountString = mount_spiner.getText().toString();
        String year2 = year_spiner.getText().toString();

        setMount(mountString);

        if (mount <= 12 && mount > 0) {
            allProductsMount(cursor, animalsType, year2);

        } else if (cursor.getCount() == 0) {
            visitors.add(new BarEntry(0, 0));
        }

        //проверка за весь год
        else if (mount == 13) {
            allProductsYear(cursor, sumCategory, year2, animalsType);
        } else {
            visitors.add(new BarEntry(0, 0));
        }
    }


    public void allProductsMount(Cursor cursor, String animalsType, String year2) {
        float x, y;
        while (cursor.moveToNext()) {
            // проверка статуса
            if (animalsType.equals(cursor.getString(1))) {
                //проверка месяца
                if (mount == Integer.parseInt(cursor.getString(4))) {
                    //проверка года
                    if (year2.equals(cursor.getString(5))) {
                        //проверка статуса
                        if (status == Integer.parseInt(cursor.getString(6))) {

                            x = Float.parseFloat(cursor.getString(2));
                            y = Float.parseFloat(cursor.getString(3));

                            visitors.add(new BarEntry(y, x));
                        }
                    }
                }
            }
        }
        cursor.close();
    }

    public void allProductsYear(Cursor cursor, Map<Float, Float> sumCategory, String year2, String animalsType) {
        sumCategory.put(1F, (float) 0);
        sumCategory.put(2F, (float) 0);
        sumCategory.put(3F, (float) 0);
        sumCategory.put(4F, (float) 0);
        sumCategory.put(5F, (float) 0);
        sumCategory.put(6F, (float) 0);
        sumCategory.put(7F, (float) 0);
        sumCategory.put(8F, (float) 0);
        sumCategory.put(9F, (float) 0);
        sumCategory.put(10F, (float) 0);
        sumCategory.put(11F, (float) 0);
        sumCategory.put(12F, (float) 0);

        while (cursor.moveToNext()) {
            if (animalsType.equals(cursor.getString(1))) {
                if (year2.equals(cursor.getString(5))) {
                    if (status == Integer.parseInt(cursor.getString(6))) {
                        float sum = sumCategory.get(Float.parseFloat(cursor.getString(4))) + Float.parseFloat(cursor.getString(2));
                        sumCategory.put(Float.parseFloat(cursor.getString(4)), sum);
                    }
                }
            }
        }
        cursor.close();

        for (Map.Entry<Float, Float> entry : sumCategory.entrySet()) {
            Float name = entry.getKey();
            Float sum = entry.getValue();

            visitors.add(new BarEntry(name, sum));
        }
    }

    public void setMount(String mountString) {
        switch (mountString) {
            case "Январь":
                mount = 1;
                mountMass = new String[]{"", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", "29", "30", "31", ""};
                break;
            case "Февраль":
                mount = 2;
                mountMass = new String[]{"", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", ""};
                break;
            case "Март":
                mount = 3;
                mountMass = new String[]{"", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", "29", "30", "31", ""};
                break;
            case "Апрель":
                mount = 4;
                mountMass = new String[]{"", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", "29", "30", ""};
                break;
            case "Май":
                mount = 5;
                mountMass = new String[]{"", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", "29", "30", "31", ""};
                break;
            case "Июнь":
                mount = 6;
                mountMass = new String[]{"", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", "29", "30", ""};
                break;
            case "Июль":
                mount = 7;
                mountMass = new String[]{"", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", "29", "30", "31", ""};
                break;
            case "Август":
                mount = 8;
                mountMass = new String[]{"", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", "29", "30", "31", ""};
                break;
            case "Сентябрь":
                mount = 9;
                mountMass = new String[]{"", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", "29", "30", ""};
                break;
            case "Октябрь":
                mount = 10;
                mountMass = new String[]{"", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", "29", "30", "31", ""};
                break;
            case "Ноябрь":
                mount = 11;
                mountMass = new String[]{"", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", "29", "30", ""};
                break;
            case "Декабрь":
                mount = 12;
                mountMass = new String[]{"", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", "29", "30", "31", ""};
                break;
            case "За весь год":
                mount = 13;
                break;
        }
    }
}

File: app/src/main/java/com/hfad/myferma/Chart/WriteOffChartFragment.kt
package com.hfad.myferma.Chart

import android.database.Cursor
import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.Fragment
import com.github.mikephil.charting.charts.BarChart
import com.github.mikephil.charting.components.XAxis
import com.github.mikephil.charting.data.BarData
import com.github.mikephil.charting.data.BarDataSet
import com.github.mikephil.charting.data.BarEntry
import com.github.mikephil.charting.formatter.IndexAxisValueFormatter
import com.github.mikephil.charting.utils.ColorTemplate
import com.google.android.material.appbar.MaterialToolbar
import com.google.android.material.bottomsheet.BottomSheetDialog
import com.google.android.material.floatingactionbutton.ExtendedFloatingActionButton
import com.hfad.myferma.Settings.InfoFragment
import com.hfad.myferma.R
import com.hfad.myferma.Settings.SettingsFragment
import com.hfad.myferma.db.MyFermaDatabaseHelper
import java.util.Calendar

class WriteOffChartFragment : Fragment() {

    private val mountClass = ChartMount()
    private lateinit var myDB: MyFermaDatabaseHelper
    private lateinit var layout: View

    private lateinit var animalsSpiner: AutoCompleteTextView
    private lateinit var mountSpiner: AutoCompleteTextView
    private lateinit var yearSpiner: AutoCompleteTextView

    private lateinit var radioGroup: RadioGroup

    private lateinit var button: Button
    private lateinit var bottomSheetDialog: BottomSheetDialog

    private val labes = mutableListOf<String>(
        "",
        "Январь",
        "Февраль",
        "Март",
        "Апрель",
        "Май",
        "Июнь",
        "Июль",
        "Август",
        "Сентябрь",
        "Октябрь",
        "Ноябрь",
        "Декабрь",
        ""
    )

    private var visitors = mutableListOf<BarEntry>()
    private var mountMass = mutableListOf<String>()

    private var mount: Int = 0
    private var mountString = "За весь год"

    private var status: Int = R.drawable.baseline_cottage_24

    private var yearList = mutableListOf<String>()
    private var productList = mutableListOf<String>()

    private var infoChart: String = "График списанной продукции на собсвенные нужды"

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        layout = inflater.inflate(R.layout.fragment_write_off_chart, container, false)

        //Подключение к базе данных
        myDB = MyFermaDatabaseHelper(requireContext())

        add()

        //убириаем фаб кнопку
        val fab: ExtendedFloatingActionButton =
            requireActivity().findViewById<View>(R.id.extended_fab) as ExtendedFloatingActionButton
        fab.visibility = View.GONE

        //настройка верхнего меню фаб кнопку
        val appBar: MaterialToolbar =
            requireActivity().findViewById<MaterialToolbar>(R.id.topAppBar)
        appBar.title = "Мои Списания - График"
        appBar.setNavigationIcon(R.drawable.baseline_arrow_back_24)
        appBar.setNavigationOnClickListener {
            requireActivity().supportFragmentManager.popBackStack()
        }
        appBar.menu.findItem(R.id.magazine).isVisible = false
        appBar.menu.findItem(R.id.filler).isVisible = true
        appBar.setOnMenuItemClickListener(Toolbar.OnMenuItemClickListener { item: MenuItem ->
            when (item.itemId) {
                R.id.filler -> bottomSheetDialog.show()
                R.id.more -> {
                    moveToNextFragment(InfoFragment())
                    appBar.title = "Информация"
                }

                R.id.setting -> {
                    moveToNextFragment(SettingsFragment())
                    appBar.title = "Мои настройки"
                }
            }
            true
        })

        showBottomSheetDialog()
        //Создание списка с данными для графиков
        val calendar = Calendar.getInstance()

        // настройка спинеров
        mountSpiner.setText(mountString, false)
        animalsSpiner.setText(productList[0], false)
        yearSpiner.setText(calendar.get(Calendar.YEAR).toString(), false)

        //Логика просчета
        storeDataInArrays()

        button.setOnClickListener {
            storeDataInArrays()
            bottomSheetDialog.dismiss()
        }
        return layout
    }


    //Добавляем bottobSheet
    private fun showBottomSheetDialog() {
        bottomSheetDialog = BottomSheetDialog(requireContext())
        bottomSheetDialog.setContentView(R.layout.fragment_bottom_chart_setting)
        animalsSpiner = bottomSheetDialog.findViewById<AutoCompleteTextView>(R.id.animals_spiner)!!
        mountSpiner = bottomSheetDialog.findViewById<AutoCompleteTextView>(R.id.mounts_spiner)!!
        yearSpiner = bottomSheetDialog.findViewById<AutoCompleteTextView>(R.id.year_spiner)!!
        radioGroup = bottomSheetDialog.findViewById<RadioGroup>(R.id.radioGroup)!!
        button = bottomSheetDialog.findViewById<Button>(R.id.add_button)!!
    }

    private fun bar(xAsis: MutableList<String>, info: String) {
        val barChart: BarChart = layout.findViewById(R.id.barChart)
        val barDataSet: BarDataSet = BarDataSet(visitors, animalsSpiner.text.toString())
        barDataSet.colors = ColorTemplate.MATERIAL_COLORS.toMutableList()
        barDataSet.valueTextColor = Color.BLACK
        barDataSet.valueTextSize = 16f
        val barData: BarData = BarData(barDataSet)
        barChart.invalidate()
        barChart.setFitBars(true)
        barChart.data = barData
        barChart.description.text = info
        barChart.animateY(2000)
        val xAxis: XAxis = barChart.xAxis
        xAxis.position = XAxis.XAxisPosition.BOTTOM
        xAxis.setDrawGridLines(false)
        xAxis.granularity = 1f // only intervals of 1 day
        xAxis.labelCount = 6 //сколько отображается
        xAxis.valueFormatter = IndexAxisValueFormatter(xAsis)
    }

    override fun onStart() {
        super.onStart()
        val view: View? = view
        if (view != null) {
            //Настройка спинера с продуктами
            val arrayAdapterProduct = ArrayAdapter<String>(
                requireContext().applicationContext,
                android.R.layout.simple_spinner_dropdown_item,
                productList
            )
            animalsSpiner.setAdapter<ArrayAdapter<String>>(arrayAdapterProduct)

            // настройка спинера с годами (выглядил как обычный, и год запоминал)
            val arrayAdapterYear = ArrayAdapter<String>(
                requireContext().applicationContext,
                android.R.layout.simple_spinner_dropdown_item,
                yearList
            )
            yearSpiner.setAdapter<ArrayAdapter<String>>(arrayAdapterYear)
        }
    }

    //читаем БД для добавления списка в спинеры
    fun add() {
        val yearSet: MutableSet<String> = HashSet()
        val productSet: MutableSet<String> = HashSet()

        val cursor: Cursor = myDB.readAllDataWriteOff()
        if (cursor.count != 0) {
            while (cursor.moveToNext()) {
                val year: String = cursor.getString(5)
                val product: String = cursor.getString(1)
                yearSet.add(year)
                productSet.add(product)
            }
        } else {
            val calendar = Calendar.getInstance()
            yearSet.add(calendar[Calendar.YEAR].toString())
            productSet.add("Нет товаров")
        }
        cursor.close()

        yearList = yearSet.toMutableList()
        productList = productSet.toMutableList()
    }


    //Логика добавления список для графиков
    private fun storeDataInArrays() {

        visitors.clear()

        val animalsType: String = animalsSpiner.text.toString()
        mountString = mountSpiner.text.toString()
        val year2: String = yearSpiner.text.toString()

        val radioChoise =
            bottomSheetDialog.findViewById<RadioButton>(radioGroup.checkedRadioButtonId)

        when (radioChoise?.text.toString()) {
            "На собственные нужды" -> {
                infoChart = "График списанной продукции на собсвенные нужды"
                status = R.drawable.baseline_cottage_24
            }

            "На утилизацию" -> {
                infoChart = "График списанной продукции на утилизацию"
                status = R.drawable.baseline_delete_24
            }
        }

        mount = mountClass.setMountInt(mountString)

        when (mount) {
            in 1..12 -> {
                allProductsMount(
                    myDB.selectChartMountWriteOff(
                        animalsType,
                        mount.toString(),
                        year2,
                        status.toString()
                    )
                )
                bar(mountMass, infoChart)
            }

            13 -> {
                allProductsYear(
                    myDB.selectChartYearWriteOff(
                        animalsType,
                        year2,
                        status.toString()
                    )
                )
                bar(labes, infoChart)
            }

            else -> {
                for (i in 0..(labes.size - 2)) {
                    visitors.add(BarEntry(0f, 0f))
                }
                bar(labes, infoChart)
            }
        }
    }

    //Считаем продукцию за месяц
    private fun allProductsMount(cursor: Cursor) {
        if (cursor.count != 0) {
            while (cursor.moveToNext()) {
                visitors.add(BarEntry(cursor.getString(1).toFloat(), cursor.getString(0).toFloat()))
            }
        } else visitors.add(BarEntry(0f, 0f))
        cursor.close()
    }

    //Считаем продукцию за год
    private fun allProductsYear(cursor: Cursor) {

        for (i in 0..12) {
            visitors.add(BarEntry(i.toFloat(), 0f))
        }

        if (cursor.count != 0) {
            while (cursor.moveToNext()) {
                when (cursor.getString(1).toInt()) {

                    1 -> visitors[0] = BarEntry(1f, cursor.getString(0).toFloat())
                    2 -> visitors[1] = BarEntry(2f, cursor.getString(0).toFloat())
                    3 -> visitors[2] = BarEntry(3f, cursor.getString(0).toFloat())
                    4 -> visitors[3] = BarEntry(4f, cursor.getString(0).toFloat())
                    5 -> visitors[4] = BarEntry(5f, cursor.getString(0).toFloat())
                    6 -> visitors[5] = BarEntry(6f, cursor.getString(0).toFloat())
                    7 -> visitors[6] = BarEntry(7f, cursor.getString(0).toFloat())
                    8 -> visitors[7] = BarEntry(8f, cursor.getString(0).toFloat())
                    9 -> visitors[8] = BarEntry(9f, cursor.getString(0).toFloat())
                    10 -> visitors[9] = BarEntry(10f, cursor.getString(0).toFloat())
                    11 -> visitors[10] = BarEntry(11f, cursor.getString(0).toFloat())
                    12 -> visitors[11] = BarEntry(12f, cursor.getString(0).toFloat())
                }
            }
        }
        cursor.close()
    }

    private fun moveToNextFragment(fragment: Fragment) {
        requireActivity().supportFragmentManager.beginTransaction()
            .replace(R.id.conteiner, fragment, "visible_fragment")
            .addToBackStack(null)
            .commit()
    }
}